sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/DecorrelateInnerQuery.scala

```diff
@@ -664,14 +664,16 @@ object DecorrelateInnerQuery extends PredicateHelper {
               decorrelate(child, newOuterReferences, aggregated = true, underSetOp)
             // For now these are no-op, as we don't allow correlated references in the window
             // function itself.
-            val newProjectList = replaceOuterReferences(projectList, outerReferenceMap)
+            val newWindowExpressions = replaceOuterReferences(windowExpressions, outerReferenceMap)
             val newPartitionSpec = replaceOuterReferences(partitionSpec, outerReferenceMap)
             val newOrderSpec = replaceOuterReferences(orderSpec, outerReferenceMap)
-            val referencesToAdd = missingReferences(newProjectList, joinCond)
+            val referencesToAdd = missingReferences(newWindowExpressions, joinCond)
 
-            val newWindow = Window(newProjectList ++ referencesToAdd,
```

Comment:
look at `Window#output`:
```
  override def output: Seq[Attribute] =
    child.output ++ windowExpressions.map(_.toAttribute)
```

We don't need to add `referencesToAdd` to propagate the added references, as it's already propagated.